### ISSUE TITLE
getting the value of a variable in php

### ISSUE THREAD

**Issue by xmrschool** [notgiven688/webminerpool#121]
Good day! I'm trying to get the value of the entire hash, for the user, in the form of a php variable, but alas, it does not work out .. alas, my knowledge in JSON is clearly not enough ...
I will be very grateful for the examples!

**Comment by VidYen** [comment]
Are you trying to see how many hashes a user mined and report it a PHP?

If so I already have a solution or at least and example..,

But which mining pool did you plan on using?

**Comment by xmrschool** [comment]
The pool, on the basis of cryptonote-nodejs-pool. Of course, you can take data on the hashes from the pool, but then there is this information here. probably faster to get this data directly from the server?

**Comment by VidYen** [comment]
Oh yeah...  I used to do that for people when I originally created the VidYen Webminer Plugin for rewards sites.

Sooo... I think the code i still around, but basically added a webserver module and used Nginx to report the hashes.

That said, it was a pain because Nginx would crash more than the webminer, and then people complained because the hashes reported by the websocket server is just hashses worked...

Not hashes accepted.

But this might be different... Personally, I prefer to use the pools API as its the truth of how many hashes were actually accepted.

Here is my AJAX API pull using WordPress which yes its using its own functions for the CURL, but hopefully you will get this idea.

https://github.com/VidYen/VidYen-WordPress-Plugins/blob/master/vidyen-point-system-vyps/includes/functions/ajax/vyps_mo_ajax.php

```php
<?php

if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly

/*** AJAX TO GRAB HASH PER SECOND FROM MO ***/

// register the ajax action for authenticated users
add_action('wp_ajax_vyps_mo_api_action', 'vyps_mo_api_action');

//register the ajax for non authenticated users
//NOTE: I need to get feed back if nonautheneticaed users should actually be able to use this.
add_action( 'wp_ajax_nopriv_vyps_mo_api_action', 'vyps_mo_api_action' );

// handle the ajax request
function vyps_mo_api_action()
{
  global $wpdb; // this is how you get access to the database

  //NOTE: I do not think there is a need for nonce as no user input to wordpress

  //Post gather from the AJAX post
  $site_wallet = sanitize_text_field($_POST['site_wallet']);
  $site_worker = sanitize_text_field($_POST['site_worker']);

  //Copy and paste from the Shortcodes
  //MO remote get info for client

  //MO remote get info for site
  $mo_site_wallet = $site_wallet;
  $mo_site_worker = $site_worker;

  /*** MoneroOcean Gets***/
  //Site get
  $site_url = 'https://api.moneroocean.stream/miner/' . $mo_site_wallet . '/stats/' . $mo_site_worker;
  $site_mo_response = wp_remote_get( $site_url );
  if ( is_array( $site_mo_response ) )
  {
    $site_mo_response = $site_mo_response['body']; // use the content
    $site_mo_response = json_decode($site_mo_response, TRUE);
    if (array_key_exists('totalHash', $site_mo_response))
    {
      $site_total_hashes = floatval($site_mo_response['totalHash']); //I'm removing the number format as we need the raw data.
      $site_valid_shares = floatval($site_mo_response['validShares']); //I'm removing the number format as we need the raw data.
      $site_hash_per_second = number_format(intval($site_mo_response['hash2'])); //We already know site total hashes.
      $site_hash_per_second = ' ' . $site_hash_per_second . ' H/s';
    }
    else
    {
      $site_hash_per_second = '';
      $site_total_hashes = 0;
      $site_valid_shares = 0; //Moving to the share system as better to predict payments.
    }
  }

  //Let's get the price of XMR now if we can:

  $current_xmr_price = vyps_mo_xmr_usd_api();

  $mo_array_server_response = array(
      'site_hashes' => $site_total_hashes,
      'site_hash_per_second' => $site_hash_per_second,
      'site_validShares' => $site_valid_shares,
      'current_XMRprice' => $current_xmr_price,
  );

  echo json_encode($mo_array_server_response); //Proper method to return json

  wp_die(); // this is required to terminate immediately and return a proper response
}
```

This IMO is the best way to handle it as we are getting accepted and not worked... But I can dig up the old code but I pretty sure its been discarded and buried under a lot of commits.

**Comment by VidYen** [comment]
Side note... The worker is actually a combination of the user name and a transaction id and MoneroOcean is fine with.

`wallet.worker` format while some pools need the worker in the password.

**Comment by sahaab** [comment]
> Side note... The worker is actually a combination of the user name and a transaction id and MoneroOcean is fine with.
> 
> `wallet.worker` format while some pools need the worker in the password.

Hey, currently, I am mining on MoneroOcean, and using their API to figure out the hashes mined by users. But for this, I have to specify a different 'password' for every user. Which technically means a new connection for every user to MoneroOceans server. So far I have around 100+ connections to MoneroOceans server and they have started to join some of my miners.
I was hoping, I would be able to use the server to differentiate users via the UserID, while keeping the same password so on MoneroOcean it shows as 1 connection.

My questions regarding this are:
1. Is this possible? E.g can I use the server, with same loginId and different userID to calculate the amount of hashes mined by the user.
2. Is there some way for my backend to communicate with the mining server to get the total number of hashes mined by the specific user? (And reward them accordingly)
3. Is there a way I can connect an xmrig app to the mining server?
